### PR TITLE
Deployer types

### PR DESCRIPTION
**Pull request by mfleader**
## Changes introduced with this PR

Adjust deployer based on parameters in workflow plugin.

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: internal/step/dummy/provider_test.go [hunk 1]
```diff
@@ -29,13 +29,13 @@ func (s *stageChangeHandler) OnStepComplete(
 		panic(fmt.Errorf("invalid previous stage: %s", previousStage))
 	}
 	if previousStageOutputID == nil {
-		panic(fmt.Errorf("no previous stage output ID"))
+		panic(fmt.Errorf("no previous stage output Name"))
```
Comment (dbutenhof) [inline internal/step/dummy/provider_test.go:32]: Is this change intentional/correct, or did it get caught in a global replace? Why would a "stage output" message change from "ID" to "Name" because of the deployer terminology change???

File: internal/step/dummy/provider_test.go [hunk 2]
```diff
@@ -29,13 +29,13 @@ func (s *stageChangeHandler) OnStepComplete(
...
 	}
 	if *previousStageOutputID != "success" {
-		panic(fmt.Errorf("invalid previous stage output ID: %s", *previousStageOutputID))
+		panic(fmt.Errorf("invalid previous stage output Name: %s", *previousStageOutputID))
```
Comment (dbutenhof) [inline internal/step/dummy/provider_test.go:35]: And here ...

File: internal/step/dummy/provider_test.go [hunk 3]
```diff
@@ -29,13 +29,13 @@ func (s *stageChangeHandler) OnStepComplete(
...
 	}
 	if previousStageOutput == nil {
-		panic(fmt.Errorf("no previous stage output ID"))
+		panic(fmt.Errorf("no previous stage output Name"))
```
Comment (dbutenhof) [inline internal/step/dummy/provider_test.go:38]: Here...

File: internal/step/foreach/provider.go [hunk 4]
```diff
@@ -471,7 +471,7 @@ func (r *runningStep) run() {
 				}
 
 				r.logger.Debugf("Executing item %d...", i)
-				// Ignore the output ID here because it can only be "success"
+				// Ignore the output Name here because it can only be "success"
```
Comment (dbutenhof) [inline internal/step/foreach/provider.go:474]: Here ...

File: internal/step/lifecycle.go [hunk 5]
```diff
@@ -49,7 +49,7 @@ func (l Lifecycle[StageType]) DAG() (dgraph.DirectedGraph[StageType], error) {
 
 // lifecycleStage is a helper interface for being able to construct a DAG from a lifecycle.
 type lifecycleStage interface {
-	// Identifier returns the ID of the stage.
+	// Identifier returns the Name of the stage.
```
Comment (dbutenhof) [inline internal/step/lifecycle.go:52]: Here ...

File: internal/step/plugin/provider.go [hunk 6]
```diff
@@ -291,11 +364,11 @@ func (r *runnableStep) Lifecycle(input map[string]any) (result step.Lifecycle[st
 	cancelSignal := stepSchema.SignalHandlers()[plugin.CancellationSignalSchema.ID()]
 	if cancelSignal == nil {
 		// Not present
-		stopIfProperty.Disable(fmt.Sprintf("Cancel signal with ID '%s' is not present in plugin image '%s', step '%s'. Signal handler IDs present: %v",
-			plugin.CancellationSignalSchema.ID(), r.image, stepID, reflect.ValueOf(stepSchema.SignalHandlers()).MapKeys()))
+		stopIfProperty.Disable(fmt.Sprintf("Cancel signal with Name '%s' is not present in plugin '%s', step '%s'. Signal handler IDs present: %v",
```
Comment (dbutenhof) [inline internal/step/plugin/provider.go:367]: `Name` or `ID`???

File: internal/step/plugin/provider_test.go [hunk 7]
```diff
@@ -32,13 +33,13 @@ func (s *deployFailStageChangeHandler) OnStepComplete(
 		panic(fmt.Errorf("invalid previous stage: %s", previousStage))
 	}
 	if previousStageOutputID == nil {
-		panic(fmt.Errorf("no previous stage output ID"))
+		panic(fmt.Errorf("no previous stage output Name"))
```
Comment (dbutenhof) [inline internal/step/plugin/provider_test.go:36]: Name? ID? Why did "stage output" terminology change?

Also 39, 42, 68, 71... and others

File: workflow/executor.go [hunk 8]
```diff
@@ -58,7 +58,7 @@ type ExecutableWorkflow interface {
 
 	// Execute runs a workflow until it finishes or until the context expires with the specified input. The input
 	// must only contain primitives (float, int, bool, string, map, slice) and may not contain structs and other
-	// elements. The output will consist of the output ID, the returned output data corresponding to the output IDs
+	// elements. The output will consist of the output Name, the returned output data corresponding to the output IDs
```
Comment (dbutenhof) [inline workflow/executor.go:61]: Name/ID?

File: workflow/model.go [hunk 9]
```diff
@@ -21,7 +21,7 @@ type Workflow struct {
 	// Steps contains the possible steps in this workflow. The data set must contain a valid step structure where the
 	// inputs to stages may consist only of primitive types and expressions.
 	Steps map[string]any `json:"steps"`
-	// Outputs lets you define one or more outputs. The outputs should be keyed by their output ID (e.g. "success") and
+	// Outputs lets you define one or more outputs. The outputs should be keyed by their output Name (e.g. "success") and
```
Comment (dbutenhof) [inline workflow/model.go:24]: Name/ID? (Plus several more in this file)

File: workflow/workflow.go [hunk 10]
```diff
@@ -260,31 +260,31 @@ func (l *loopState) onStageComplete(stepID string, previousStage *string, previo
 	}
 	stageNode, err := l.dag.GetNodeByID(GetStageNodeID(stepID, *previousStage))
 	if err != nil {
-		l.logger.Errorf("Failed to get stage node ID %s (%w)", GetStageNodeID(stepID, *previousStage), err)
-		l.recentErrors <- fmt.Errorf("failed to get stage node ID %s (%w)", GetStageNodeID(stepID, *previousStage), err)
+		l.logger.Errorf("Failed to get stage node Name %s (%w)", GetStageNodeID(stepID, *previousStage), err)
```
Comment (dbutenhof) [inline workflow/workflow.go:263]: stage node Name or stage node ID??? (and also more below)

File: workflow/model.go (hunk 9 above)
Comment (jaredoconnell) [inline workflow/model.go]: Out of scope change

File: workflow/model.go [hunk 11]
```diff
@@ -182,7 +182,7 @@ type DAGItem struct {
 	StepID string
 	// StageID is the stage of the step provider this item refers to.
 	StageID string
-	// OutputID is the ID of the output of the step stage.
+	// OutputID is the Name of the output of the step stage.
```
Comment (jaredoconnell) [inline workflow/model.go]: Out of scope change

File: workflow/model.go [hunk 12]
```diff
@@ -212,12 +212,12 @@ func (d DAGItem) String() string {
...
 }
 
-// GetStageNodeID returns the DAG node ID for a stage.
+// GetStageNodeID returns the DAG node Name for a stage.
 func GetStageNodeID(stepID string, stageID string) string {
 	return fmt.Sprintf("steps.%s.%s", stepID, stageID)
 }
 
-// GetOutputNodeID returns the DAG node ID for a stage output.
+// GetOutputNodeID returns the DAG node Name for a stage output.
```
Comment (jaredoconnell) [inline workflow/model.go]: Out of scope changes

**Review by jaredoconnell — CHANGES_REQUESTED** [review]
The name/ID refactor made changes out of scope.
In addition, we should probably add a test for invalid deployer ID. I'll do that.

**Review by jaredoconnell — APPROVED** [review]
Looks good to me. Tested locally.
